routing_common/routing_common_tests/vehicle_model_for_country_test.cpp

```diff
@@ -0,0 +1,151 @@
+#include <string>
+
+using namespace std;
+using namespace routing;
+
+namespace
+{
+string GetRegionParent(string const & id)
+{
+  if (id == "Moscow")
+    return "Russian Federation";
+  if (id == "Munich")
+    return "Bavaria";
+  if (id == "Bavaria")
+    return "Germany";
+  if (id == "San Francisco")
+    return "California";
+  if (id == "California")
+    return "United States of America";
+  return "";
+}
+
+template <typename VehicleModelType, typename VehicleModelFactoryType>
+void TestVehicleModelDefault()
+{
+  auto defaultVehicleModel = make_shared<VehicleModelType>();
+
+  VehicleModelFactoryType vehicleModelFactory = VehicleModelFactoryType(GetRegionParent);
+
+  // To not add virtual EqualsForTests to VehicleModelInterface, CrossCountryVehicleModel etc
+  shared_ptr<VehicleModelType> defaultVehicleModelForCountry = static_pointer_cast<VehicleModelType>(
+      vehicleModelFactory.GetVehicleModelForCountry("Nonexistent Country Name"));
+  TEST(defaultVehicleModel->EqualsForTests(*defaultVehicleModelForCountry),
+       ("Vehicle model for nonexistent counry is not equal to default."));
+}
+
+// DirectParent and IndirectParent tests require tested countries to have nondefault restriction
+// If selected countries will change they restrictions to default select other countries for tests
+template <typename VehicleModelType, typename VehicleModelFactoryType>
+void TestHaveNondefaultRestrictionForSelectedCountry(string country)
+{
+  auto defaultVehicleModel = make_shared<VehicleModelType>();
+
+  VehicleModelFactoryType vehicleModelFactory = VehicleModelFactoryType(GetRegionParent);
+
+  shared_ptr<VehicleModelType> vehicleModelCountry =
+      static_pointer_cast<VehicleModelType>(vehicleModelFactory.GetVehicleModelForCountry(country));
+
+  TEST(!(vehicleModelCountry->EqualsForTests(*defaultVehicleModel)),
+       (country,
+        "has default model. It may be ok if trafic restrictions was changed. "
+        "If so, select other country for this and next test."));
+}
+
+template <typename VehicleModelType, typename VehicleModelFactoryType>
+void ParentTest(string child, string parent)
+{
+  VehicleModelFactoryType vehicleModelFactory = VehicleModelFactoryType(GetRegionParent);
+
+  shared_ptr<VehicleModelType> vehicleModelChild =
+      static_pointer_cast<VehicleModelType>(vehicleModelFactory.GetVehicleModelForCountry(child));
+  shared_ptr<VehicleModelType> vehicleModelParent =
+      static_pointer_cast<VehicleModelType>(vehicleModelFactory.GetVehicleModelForCountry(parent));
+
+  TEST(vehicleModelChild->EqualsForTests(*vehicleModelParent),
+       ("Can not expand car model for", child, "to", parent));
+}
+}  // namespace
+
+// Test we have default vehicle models for nonexistend(unknown) country
+UNIT_TEST(CarModel_Default)
+{
+  classificator::Load();
+
+  TestVehicleModelDefault<CarModel, CarModelFactory>();
+}
+
+UNIT_TEST(BicycleModel_Default)
+{
+  TestVehicleModelDefault<BicycleModel, BicycleModelFactory>();
```

Comment:
Seems that you need to load classificator in all tests. We have "--filter" flag in tests binaries, so if someone'll run "BicycleModel_Default" test, it'll fail. To prevent this, define a ModelTest class which loads classificator in ctor, and use UNIT_CLASS_TEST for all tests here.

Reply:
done